tickit/devices/eiger/eiger.py

```diff
@@ -0,0 +1,453 @@
+
+        If the detector is in an external trigger mode, this is disabled as
+        this software command interface only works for internal triggers.
+        """
+        trigger_mode = self.settings.trigger_mode
+        state = self.status.state
+
+        if state == State.READY and trigger_mode == "ints":
+            self._set_state(State.ACQUIRE)
+
+            for idx in range(0, self.settings.nimages):
+
+                aquired = Image.create_dummy_image(idx)
+
+                header_json = {
+                    "htype": "dimage-1.0",
+                    "series": "<series id>",
+                    "frame": aquired.index,
+                    "hash": aquired.hash,
+                }
+
+                json2 = {
+                    "htype": "dimage_d-1.0",
+                    "shape": "[x,y,(z)]",
+                    "type": aquired.dtype,
+                    "encoding": aquired.encoding,
+                    "size": len(aquired.data),
+                }
+
+                json3 = {
+                    "htype": "dconfig-1.0",
+                    "start_time": "<start_time>",
+                    "stop_time": "<stop_time>",
+                    "real_time": "<real_time>",
+                }
+
+                LOGGER.debug(header_json)
+                LOGGER.debug(json2)
+                LOGGER.debug(json3)
+
+            return "Aquiring Data from Eiger..."
+        else:
+            return (
+                f"Ignoring trigger, state={self.status.state},"
+                f"trigger_mode={trigger_mode}"
+            )
+
+    async def cancel(self) -> None:
+        """Function to stop the data acquisition.
+
+        Function to stop the data acquisition, but only after the next
+        image is finished.
+        """
+        self._set_state(State.READY)
+
+        header_json = {"htype": "dseries_end-1.0", "series": "<id>"}
+
+        LOGGER.debug(header_json)
+
+    async def abort(self) -> None:
+        """Function to abort the current task on the Eiger."""
+        self._set_state(State.IDLE)
+
+        header_json = {"htype": "dseries_end-1.0", "series": "<id>"}
+
+        LOGGER.debug(header_json)
+
+    def update(self, time: SimTime, inputs) -> DeviceUpdate:
+        """Generic update function to update the values of the ExampleHTTPDevice.
+
+        Args:
+            time (SimTime): The simulation time in nanoseconds.
+            inputs (Inputs): A TypedDict of the inputs to the ExampleHTTPDevice.
+
+        Returns:
+            DeviceUpdate[Outputs]:
+                The produced update event which contains the value of the device
+                variables.
+        """
+        current_flux = inputs["flux"]
+        # if current_flux > 0:
+        # TODO: Figure out how to scale the output image
+        intensity_scale = (current_flux / 100) * 100
+        LOGGER.debug(f"Relative beam intensity: {intensity_scale}")
+
+        return DeviceUpdate(self.Outputs(), SimTime(time + int(1e8)))
```

Comment:
Why do we update every 0.1s?

Reply:
While a fixed cycle is probably an okay start, I think the eiger should eventually be purely event-based

Reply:
Considering what's contained within `update` at the moment, there's no reason not to be purely event-based as is.

Also, I'm not sure I can envisage any scenarios where a device without outputs should be asking for a callback, perhaps we should warn when this is done?

Reply:
it was event based before, but then it would cause the simulation to crash as it would interrupt before it had effectively done a tick

Reply:
Seems to work fine for me

Reply:
So think I should  remove the SimTime?

Reply:
Reckon so